components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/UserSessionStore.java

```diff
@@ -428,16 +462,143 @@ private void deleteSessionToUserMappingsFor(Set<String> terminatedSessionIds, Co
+     *
+     * @param sessionId   id of the authenticated session
+     * @param subject     username in application
+     * @param appID       id of the application
+     * @param inboundAuth protocol used in app
+     * @throws DataAccessException if an error occurs when storing the authenticated user details to the database
+     */
+    public void storeAppSessionData(String sessionId, String subject, int appID, String inboundAuth) throws
+            DataAccessException {
+
+        if (appID != 0) {
+            JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+            try {
+                jdbcTemplate.executeUpdate(SQLQueries.SQL_STORE_IDN_AUTH_SESSION_APP_INFO, preparedStatement -> {
+                    preparedStatement.setString(1, sessionId);
+                    preparedStatement.setString(2, subject);
+                    preparedStatement.setInt(3, appID);
+                    preparedStatement.setString(4, inboundAuth);
+                });
+            } catch (DataAccessException e) {
+                throw new DataAccessException("Error while storing application data for session in the database ", e);
+            }
         }
     }
 
+    /**
+     * Method to get app id from SP_APP table.
+     *
+     * @param applicationName application Name
+     * @param appTenantID     app tenant id
+     * @return the application id
+     * @throws UserSessionException if an error occurs when retrieving app id
+     */
+    public int getAppId(String applicationName, int appTenantID) throws UserSessionException {
+
+        final int[] applicationId = {0};
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        try {
+            jdbcTemplate.fetchSingleRecord(SQLQueries.SQL_SELECT_APP_ID_OF_APP,
+                    ((resultSet, rowNumber) -> applicationId[0] = resultSet.getInt(1)),
+                    preparedStatement -> {
+                        preparedStatement.setString(1, applicationName);
+                        preparedStatement.setInt(2, appTenantID);
+                    });
+        } catch (DataAccessException e) {
+            throw new UserSessionException("Error while retrieving the app id of " + applicationName + ", " +
+                    "tenant id" + appTenantID, e);
+        }
+        return applicationId[0];
+    }
+
+    /**
+     * Method to check whether the particular app session is already exists in the database.
+     *
+     * @param sessionId   id of the authenticated session
+     * @param subject     user name of app
+     * @param appID       id of application
+     * @param inboundAuth protocol used in app
+     * @return whether the app session is already available or not
+     * @throws UserSessionException while retrieving existing session data
+     */
+    public boolean isExistingAppSession(String sessionId, String subject, int appID, String inboundAuth) throws
+            UserSessionException {
+
+        final boolean[] isExisting = {false};
```

Comment:
isExisting is an array, any particular reason ?

Reply:
These fixes are from https://github.com/wso2/carbon-identity-framework/pull/2234. Will check these again and update the PR.

Reply:
Updated